Moe gemm activation

lalala-sh opened the pull request:
## Proposed changes

1. Fuse activation and moe ffn1. 
    Supported activations :
      0: gelu and mul, 
      1: silu and mul, 
2. enable 16x16 xdl size.
## Checklist

Please put an `x` into the boxes that apply. You can also fill these out after creating the PR. If you're not sure, please don't hesitate to ask.

- [x] I have added tests relevant to the introduced functionality, and the unit tests are passing locally
- [x] I have added the test to REGRESSION_TESTS list defined at the top of CMakeLists.txt in tests/CMakeLists.txt, **IF** the test takes more than 30 seconds to run.
- [x] I have added inline documentation which enables the maintainers with understanding the motivation
- [x] I have removed the stale documentation which is no longer relevant after this pull request
- [x] (If this change is user-facing) I have added release notes which provide the end users with a brief summary of the improvement from this pull request
- [x] I have run `clang-format` on all changed files
- [x] Any dependent changes have been merged

## Discussion

If this is a relatively large or complex change, feel free to start a discussion by explaining why you chose the solution you did and what alternatives you considered

coderfeli committed:
fix useless code and remove usless oob

coderfeli committed:
clang format

coderfeli committed:
Merge branch 'develop' into ck_moe_rm_oob

coderfeli committed:
fix coredump in e2e test

coderfeli committed:
fix2

illsilin committed:
fix clang format

coderfeli committed:
fix output oob

coderfeli committed:
impl int64 but result not correct

coderfeli committed:
int64 index ok now

coderfeli committed:
input output all ok

coderfeli committed:
fix uint32

coderfeli committed:
mork to support 13w tokens

coderfeli committed:
revert v1 test

coderfeli committed:
use uint32

coderfeli committed:
moe sorting fix moebuf

coderfeli committed:
merge develop

coderfeli committed:
fix merge

coderfeli committed:
update moe api fix aiter build

coderfeli committed:
fix buid

coderfeli committed:
fuse silu

coderfeli committed:
silu ok

coderfeli committed:
acale ok

coderfeli committed:
add silu

coderfeli committed:
change code

coderfeli committed:
gemm2 ok

coderfeli committed:
gufusion compatible ok, fix warnings

coderfeli committed:
gu fusion for m32 m64 ok

coderfeli committed:
support bf16 cshuffle

coderfeli committed:
i4 gemm2 ok

coderfeli committed:
i4 gemm2 ok and i4 gemm1 build

coderfeli committed:
16x16 run ok

coderfeli committed:
change flops; change cshuffle dtype

Commit:
fuse gelu silu act in moe gemm1

lalala-sh committed:
Merge remote-tracking branch 'origin/develop' into moe_gemm_activation

Commit:
fp8 with act ready

Commit:
int4 act ready

Commit:
Merge remote-tracking branch 'origin/develop' into moe_gemm_activation

Commit:
remove useless changes

lalala-sh committed:
remove useless code change

spolifroni-amd reviewed (commented):
Doesn't look like there's anything for docs.

illsilin committed:
fix clang format

illsilin commented:
Fixed clang format.
Please remember to run the script before creating PR.

Commit:
add the arch limit of int4 moe gemm

illsilin committed:
Merge branch 'develop' into moe_gemm_activation

Commit:
fuse moe activation

lalala-sh committed:
fuse moe ffn1 and activation op

1. fused moe related activation op.
2. inserted act_op parameters to DeviceMoeGemm.
  - 0: gelu
  - 1: silu
  - 2: swiglu
3. fixed moe gemm example.

Commit:
fix fp8 16x16

lalala-sh committed:
fix fp8 16x16

fix fp8 16x16

lalala-sh committed:
fix no quant case

lalala-sh committed:
fix moe no quant case

fix no quant case

Commit:
fix bugs

lalala-sh committed:
fix moe gemm1 act bugs

lalala-sh committed:
Merge branch 'develop' into moe_gemm_activation

lalala-sh committed:
merge from testx

lalala-sh committed:
Merge branch 'develop' into moe_gemm_activation

lalala-sh committed:
Merge branch 'develop' into moe_gemm_activation

lalala-sh committed:
fix fp8 gufusion bug

lalala-sh committed:
remove useless comments

lalala-sh committed:
refine activation code & complete moe example

lalala-sh committed:
Merge branch 'develop' into moe_gemm_activation

lalala-sh committed:
fix int8 bugs

lalala-sh committed:
Merge branch 'develop' into moe_gemm_activation

lalala-sh committed:
merge tkw1

lalala-sh committed:
merge from develop

coderfeli commented on example/65_gemm_multiply_multiply/moe_gemm1_xdl_pk_i4.cpp:
```diff
@@ -57,38 +58,28 @@ struct MulABScale
+        (void)d1;
 #if CK_USE_PK4_LAYOUT_SHUFFLE
-        e = ck::type_convert<EDataType>(c * d1 * d0 * 16);
+        e = ck::type_convert<EDataType>(c);
```
why not * 16 here?

lalala-sh replied:
（*16）operation has already moved into gemm pipeline.

lalala-sh committed:
Merge branch 'develop' into moe_gemm_activation

lalala-sh committed:
Merge branch 'develop' into moe_gemm_activation

lalala-sh committed:
Merge branch 'develop' into moe_gemm_activation

carlushuang reviewed (approved):
LGTM
